streamer/pulp/streamer/server.py

```diff
@@ -53,12 +55,16 @@ def __init__(self, request, streamer_config):
         :type  request:         twisted.web.server.Request
         :param streamer_config: The configuration for this streamer instance.
         :type  streamer_config: ConfigParser.SafeConfigParser
+        :param catalog_entry:   Catalog entry for the file requested.
+        :type  catalog_entry:   pulp.server.db.model.LazyCatalogEntry
+        :param pulp_request:    True if this request originated from Pulp.
```

Comment:
A request originates from outside of Pulp when a client requests a piece of content. If Pulp itself is requesting content (to save to permanent storage) it sets an HTTP header so the streamer doesn't create a database entry so that Pulp later goes and downloads the content, which causes a database entry...